main/modes/games/2048/T48_leds.c

```diff
@@ -0,0 +1,229 @@
+/**
+ * @file T48_leds.c
+ * @author Jeremy Stintzcum (jeremy.stintzcum@gmail.com)
+ * @brief LED helpers for 2048
+ * @version 1.0.0
+ * @date 2024-08-16
+ *
+ * @copyright Copyright (c) 2024
+ *
+ */
+
+#include "T48_leds.h"
+
+void t48DimLEDs(t48_t* t48)
+{
+    for (uint8_t i = 0; i < CONFIG_NUM_LEDS; i++)
+    {
+        // Red
+        if (t48->leds[i].r < 6)
+        {
+            t48->leds[i].r = 0;
+        }
+        else if (t48->leds[i].r == 0)
+        {
+            // Do nothing
+        }
+        else
+        {
+            t48->leds[i].r -= 6;
+        }
+        // Green
+        if (t48->leds[i].g < 6)
+        {
+            t48->leds[i].g = 0;
+        }
+        else if (t48->leds[i].g == 0)
+        {
+            // Do nothing
+        }
+        else
+        {
+            t48->leds[i].g -= 6;
+        }
+        // Blue
+        if (t48->leds[i].b < 6)
+        {
+            t48->leds[i].b = 0;
+        }
+        else if (t48->leds[i].b == 0)
+        {
+            // Do nothing
+        }
+        else
+        {
+            t48->leds[i].b -= 6;
+        }
+    }
+    setLeds(t48->leds, CONFIG_NUM_LEDS);
+}
+
+void t48SetRGB(t48_t* t48, uint8_t idx, led_t color)
+{
+    t48->leds[idx] = color;
+}
+
+void t48LightLEDs(t48_t* t48, t48Direction_t dir, led_t color)
+{
+    switch (dir)
+    {
+        case DOWN:
+            t48SetRGB(t48, 0, color);
+            t48SetRGB(t48, 1, color);
+            t48SetRGB(t48, 6, color);
+            t48SetRGB(t48, 7, color);
+            break;
+        case UP:
+            t48SetRGB(t48, 2, color);
+            t48SetRGB(t48, 3, color);
+            t48SetRGB(t48, 4, color);
+            t48SetRGB(t48, 5, color);
+            break;
+        case LEFT:
+            t48SetRGB(t48, 0, color);
+            t48SetRGB(t48, 1, color);
+            t48SetRGB(t48, 2, color);
+            t48SetRGB(t48, 3, color);
+            break;
+        case RIGHT:
+            t48SetRGB(t48, 4, color);
+            t48SetRGB(t48, 5, color);
+            t48SetRGB(t48, 6, color);
+            t48SetRGB(t48, 7, color);
+            break;
+        case ALL:
+            t48SetRGB(t48, 0, color);
+            t48SetRGB(t48, 1, color);
+            t48SetRGB(t48, 2, color);
+            t48SetRGB(t48, 3, color);
+            t48SetRGB(t48, 4, color);
+            t48SetRGB(t48, 5, color);
+            t48SetRGB(t48, 6, color);
+            t48SetRGB(t48, 7, color);
+            break;
+    }
+}
+
+led_t t48GetLEDColors(t48_t* t48)
```

Comment:
Be aware that this will likely need tweaking when you get hardware in-hand (unless you checked it on a Gunship already). Emulated LED colors aren't super close to actual color...